integration-libs/punchout/core/facade/punchout.service.ts

```diff
@@ -67,55 +73,44 @@ export class PunchoutService implements PunchoutFacade {
       this.displayErrorPage();
       return throwError(() => new Error('Punchout Session Id missing'));
     }
-    return this.punchoutConnector
-      .getPunchoutSession(payload.punchoutSessionId)
-      .pipe(
-        map((punchoutSession) => {
+
+    return this.requestPunchoutSession(payload.punchoutSessionId).pipe(
+      switchMap((punchoutSession) => {
+        return forkJoin({
+          punchoutSession: of(punchoutSession),
+          logout: this.punchoutAuthService.logout(),
+        });
+      }),
+      map(({ punchoutSession }) => {
+        if (punchoutSession?.token?.accessToken) {
+          this.punchoutAuthService.loginWithToken(
+            punchoutSession.token.accessToken,
+            punchoutSession.customerId
+          );
+          this.loadCart(punchoutSession.cartId).subscribe();
```

Comment:
I know it's already existing code, but nested subscription seems potentially unpredictable

Reply:
thanks,
I'm adding this concerns to 
[CXSPA-9871](https://jira.tools.sap/browse/CXSPA-9871)
Token not revoked when silently logout user
this pipe will need to be enhanced.